src/client/Microsoft.Identity.Client/ManagedIdentity/ScopeUtilities.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal static class ScopeUtilities
+    {
+        private const string DefaultSuffix = "/.default";
+        private const string ScopePattern = "^[0-9a-zA-Z-_.:/]+$";
+
+        private const string InvalidScopeMessage = "The specified scope is not in expected format. Only alphanumeric characters, '.', '-', ':', and '/' are allowed";
+        private static readonly Regex scopeRegex = new Regex(ScopePattern);
+
+        public static string ScopesToResource(string[] scopes)
+        {
+            if (scopes == null)
+            {
+                throw new ArgumentNullException(nameof(scopes));
+            }
+
+            if (scopes.Length != 1)
+            {
+                throw new ArgumentException("To convert to a resource string the specified array must be exactly length 1", nameof(scopes));
```

Comment:
same, should be in MsalErrorMessage

Reply:
It should be an MsalClientException, not an ArgumentException. Exceptions need to be actionable by app developers, this one is not.

Reply:
Updated.